pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -1328,23 +1334,41 @@ impl Timeline {
                         {
                             critical!("missing key during compaction: {err:?}");
                         }
-                    })?;
-
-                self.last_image_layer_creation_status
-                    .store(Arc::new(outcome.clone()));
-
-                self.upload_new_image_layers(image_layers)?;
-                if let LastImageLayerCreationStatus::Incomplete { .. } = outcome {
-                    // Yield and do not do any other kind of compaction.
-                    info!(
-                        "skipping shard ancestor compaction due to pending image layer generation tasks (preempted by L0 compaction)."
-                    );
-                    return Ok(CompactionOutcome::YieldForL0);
-                }
-            }
+                    });
 
-            Ok(_) => {
-                info!("skipping repartitioning due to image compaction LSN being below GC cutoff");
+                match res {
+                    Ok((image_layers, outcome)) => {
+                        self.last_image_layer_creation_status
+                            .store(Arc::new(outcome.clone()));
+
+                        self.upload_new_image_layers(image_layers)?;
+                        if let LastImageLayerCreationStatus::Incomplete { .. } = outcome {
+                            // Yield and do not do any other kind of compaction.
+                            info!(
+                                "skipping shard ancestor compaction due to pending image layer generation tasks (preempted by L0 compaction)."
+                            );
+                            return Ok(CompactionOutcome::YieldForL0);
+                        }
+                        // Fall through to shard ancestor compaction
+                    }
+                    Err(err) if lsn <= gc_cutoff => {
+                        if let CreateImageLayersError::GetVectoredError(_) = err {
+                            warn!(
+                                "could not create image layers due to {}; this is not critical because the requested image LSN is below the GC curoff",
+                                err
+                            );
```

Comment:
I would advise against moving errors in the middle, it makes sense in code with the format specifier +syntax highlighting but it's just word mash when printed at runtime

```suggestion
                            warn!(
                                "failed to create image layers but lsn is <= gc_cutoff, therefore this can happen: lsn={lsn} gc_cutoff={gc_cutoff}: {err}",
                            );
```